src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerEditCase.java

```diff
@@ -145,28 +144,37 @@ private void initialiseFlags(CaseData data) {
             );
         }
 
-        if (Objects.isNull(data.getApplicantFlags())
-            || (null != data.getApplicantFlags() && Objects.isNull(data.getApplicantFlags().getDetails()))) {
-            if (null != data.getCicCase().getApplicantFullName()) {
-                data.setApplicantFlags(Flags.builder()
+        updateApplicantFlags(data);
+        updateRepresentativeFlags(data);
+    }
+
+    private void updateRepresentativeFlags(CaseData data) {
```

Comment:
Since these 2 methods are quite similar can't we make a common updateFlags method and pass the required unique info for applicant/representative as parameters?

Reply:
As discussed, this would be refactored as separate tech-debt ticket.